lib/otdoa_al/otdoa_http.h

```diff
@@ -0,0 +1,273 @@
+
+enum {
+	FOREACH_OTDOA_HTTP_MSG(GENERATE_ENUM)
+};
+
+typedef struct OTDOA_HTTP_BLACKLIST {
+	unsigned int ecgi;
+	int age;
+} tOTDOA_HTTP_BLACKLIST;
+
+typedef struct OTDOA_HTTP_REQ {
+	unsigned int uRecommendedDelay;
+	unsigned int uEcgi;
+	unsigned int uDlearfcn;
+	uint16_t uMCC;
+	uint16_t uMNC;
+	uint16_t uPCI;
+	unsigned int uRadius;
+	unsigned int uNumCells;
+} tOTDOA_HTTP_REQ;
+
+typedef union {
+	tOTDOA_MSG_HTTP_HEADER header;
+	tOTDOA_MSG_HTTP_GET_UBSA http_get_ubsa;
+	tOTDOA_MSG_HTTP_GET_CFG http_get_cfg;
+	tOTDOA_MSG_HTTP_UPLOAD_VECTOR http_upload_vector;
+	tOTDOA_MSG_HTTP_UPLOAD_LOG http_upload_log;
+	tOTDOA_MSG_HTTP_UPLOAD_RESULTS http_upload_results;
+	tOTDOA_MSG_HTTP_REBIND http_rebind;
+	tOTDOA_MSG_HTTP_TEST_JWT http_test_jwt;
+} tOTDOA_HTTP_MESSAGE;
+
+/* HTTPS Thread Members */
+typedef struct OTDOA_HTTP_MEMBERS {
+	bool bDownloadComplete;
+	int nRange;	            /* Start offset for current request */
+	int nRangeSegmentEnd;   /* end of current segment */
+	int nRangeMax;          /* HTTPS_RANGE_MAX_DEFAULT indicates unknown size */
+	int nHeaderLength;      /* Length of header */
+	int nContentLength;     /* Length of content (body) */
+	int nBSARequests;       /* Number of BSA downloads this session */
+	bool bDisableTLS;       /* true to disable TLS and use HTTP (for uBSA DL only) */
+	bool bSkipAuth;         /* true to skip authentication when DL the uBSA */
+	uint8_t pubkey[PUBKEY_LMAX];
+	uint8_t iv[IV_LMAX];
+
+	char *pData;
+	char *csBuffer;
+	char *szSend;
+	size_t nOff;
+	int fdSocket;
+	struct addrinfo *res;
+	char szModemAddress[HELLAPHY_IP_ADDR_LMAX];
+	char szServerAddress[HELLAPHY_IP_ADDR_LMAX];
+	char ubsa_token[UBSA_TOKEN_LMAX];
+	tOTDOA_HTTP_REQ req;
+	long prsID;
+
+	tOTDOA_HTTP_BLACKLIST blacklist[BLACKLIST_SIZE];
+} tOTDOA_HTTP_MEMBERS;
+
+/* Private members */
+int http_bind(tOTDOA_HTTP_MEMBERS *pG, const char *pURL);
+int http_connect(tOTDOA_HTTP_MEMBERS *pG, const char *tls_host);
+int http_disconnect(tOTDOA_HTTP_MEMBERS *pG);
+void http_unbind(tOTDOA_HTTP_MEMBERS *pG);
+int http_format_request(tOTDOA_HTTP_MEMBERS *pG);
+int http_send_request(tOTDOA_HTTP_MEMBERS *pG, size_t n);
+int http_show_response(void);
+int http_write_chunk(tOTDOA_HTTP_MEMBERS *pG);
+
+/* Public members */
+void otdoa_http_init(void);
+int otdoa_http_handle_message(tOTDOA_HTTP_MESSAGE *pMsg);
+void otdoa_disable_tls(bool bDisableTLS);
+void otdoa_skip_config_dl(bool bSkipConfigDL);
+bool otdoa_get_skip_config_dl(void);
+bool otdoa_get_disable_tls(void);
+void otdoa_http_disable_encrypt(bool bDisableEncryption);
+bool otdao_http_get_encryption_disable(void);
+void otdoa_http_override_auth_resp(int override);
+int otdoa_http_get_override_auth_resp(void);
+
+int otdoa_http_send_log_upload(const char *vf, uint32_t repeat);
+int otdoa_http_send_vector_upload(const char *vf, uint32_t repeat);
+int otdoa_http_send_results_upload(const char *pURL, otdoa_api_results_t *pResults,
+				   const char *p_notes, const char *p_true_lat,
```

Comment:
Also a general note, Hungarian notation is not allowed for NCS code, so no `p_` prefixes for pointers etc.